src/math/power.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+ /* Include Files */
+#include <sof/math/power.h>
+
+/* Function Declarations */
+static int32_t div_s32s64(int64_t denominator);
+
+/* Function Definitions */
+/**
+ * Arguments	: int64_t denominator
+ *
+ * Return Type	: int32_t
+ */
+static inline int32_t div_s32s64(int64_t denominator)
+{
+	uint64_t b_denominator;
+	int32_t quotient;
+
+	if (denominator == 0) {
+		quotient = INT32_MAX;
+	} else {
+		if (denominator < 0)
+			b_denominator = ~(uint64_t)denominator + 1;
+		else
+			b_denominator = (uint64_t)denominator;
+
+		b_denominator = 1125899906842624ULL / b_denominator;
+		if (denominator < 0L)
+			quotient = (int32_t)-(int64_t)b_denominator;
```

Comment:
this looks weird, please explain

Reply:
Please see the code and check condition `denominator < 0`  when base value is negative.

The logic `[quotient = (int32_t)-(int64_t)b_denominator]` keeps the quotient negative when typecasting uint64_t to int32_t `quotient` for negative bases

![image](https://user-images.githubusercontent.com/61411947/140076366-3fcda839-eeab-4c10-a74e-55f949d23749.png)

Reply:
Sorry, i meant in the code, yes this makes sense, it just looks very strange on its own

Reply:
Thanks !! Yup, It's kind of typecast

Reply:
Where is the comment explaining the casting is for truncating?

Reply:
my mistake, I missed it. I will add

Reply:
thanks for adding it, but indentation is off on the comment

Reply:
Sure, updated the indentation.